src/decoders/cr2.rs

```diff
@@ -142,7 +142,8 @@ impl<'a> Decoder for Cr2Decoder<'a> {
       }
     };
 
-    let mut img = RawImage::new(camera, width, height, try!(self.get_wb(camera)), image);
+    let wb = self.get_wb(&camera)?;
+    let mut img = RawImage::new(camera, width, height, wb, image);
```

Comment:
I think it is because each argument is evaluated in order. The first one takes ownership of `camera` and then the white balance argument tried to borrow a reference, but it was already owned elsewhere. When reverting the change to make the first argument take a reference I can revert this line too.